src/hb-ot-shape-complex-arabic-table.hh

```diff
@@ -139,28 +139,28 @@ joining_type (hb_codepoint_t u)
   switch (u >> 12)
   {
     case 0x0u:
-      if (hb_in_range (u, 0x0600u, 0x08E2u)) return joining_table[u - 0x0600u + joining_offset_0x0600u];
+      if (hb_in_range<hb_codepoint_t> (u, 0x0600u, 0x08E2u)) return joining_table[u - 0x0600u + joining_offset_0x0600u];
```

Comment:
If I change 0x0600u to 0x0600uL it works arm-none-eabi but not on my default build system, so best way here would be disambiguating it I'd guess.